SLAM/iterative_closest_point/iterative_closest_point.py

```diff
@@ -35,15 +36,24 @@ def icp_matching(previous_points, current_points):
         count += 1
 
         if show_animation:  # pragma: no cover
-            plt.cla()
             # for stopping simulation with the esc key.
             plt.gcf().canvas.mpl_connect(
                 'key_release_event',
                 lambda event: [exit(0) if event.key == 'escape' else None])
-            plt.plot(previous_points[0, :], previous_points[1, :], ".r")
-            plt.plot(current_points[0, :], current_points[1, :], ".b")
-            plt.plot(0.0, 0.0, "xr")
-            plt.axis("equal")
+            if previous_points.shape[0] == 3:
+                fig = plt.figure()
+                ax = fig.add_subplot(111, projection='3d')
```

Comment:
After initialization, the figure should be clean up like:
```suggestion
                plt.cla()
```